webhook/resources/setting/validator.go

```diff
@@ -0,0 +1,50 @@
+package setting
+
+import (
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	werror "github.com/longhorn/longhorn-manager/webhook/error"
+)
+
+type settingValidator struct {
+	admission.DefaultValidator
+}
+
+func NewValidator() admission.Validator {
+	return &settingValidator{}
+}
+
+func (v *settingValidator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "settings",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.Setting{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (v *settingValidator) Create(request *admission.Request, newObj runtime.Object) error {
+	return validateSetting(newObj)
+}
+
+func (v *settingValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	return validateSetting(newObj)
```

Comment:
Need to refuse modifications happens on read-only settings. The function you are using is for **init** settings.

Reply:
But the RESTful API setting calls the `validateSetting` function as well.
- https://github.com/longhorn/longhorn-manager/blob/3163c45/api/setting.go#L46-L62
- https://github.com/longhorn/longhorn-manager/blob/3163c45/manager/setting.go#L42-L43

If I rename the function `ValidateInitSetting` to `ValidateSetting`, would it be better?

Reply:
OK. Don't you prefer to add a more restricted limitation based on the read-only field?
And I just realized that you didn't use function https://github.com/longhorn/longhorn-manager/blob/023f9f3bc704b867984de5d47a70663be93a6407/datastore/longhorn.go#L93:21

Reply:
Thanks for the reminder, I did not consider the read-only field before.

I add it to verify the read-only field modification should be refused.
Also, change to use the function https://github.com/longhorn/longhorn-manager/blob/023f9f3bc704b867984de5d47a70663be93a6407/datastore/longhorn.go#L93:21.